scalar/decimal.go

```diff
@@ -0,0 +1,145 @@
+package scalar
+
+import (
+	"github.com/apache/arrow/go/v13/arrow"
+	"github.com/apache/arrow/go/v13/arrow/decimal128"
+	"github.com/apache/arrow/go/v13/arrow/decimal256"
+)
+
+type Decimal256 struct {
+	Valid bool
+	Value decimal256.Num
+}
+
+func (s *Decimal256) IsValid() bool {
+	return s.Valid
+}
+
+func (*Decimal256) DataType() arrow.DataType {
+	return &arrow.Decimal256Type{}
+}
+
+func (s *Decimal256) Equal(rhs Scalar) bool {
+	if rhs == nil {
+		return false
+	}
+	r, ok := rhs.(*Decimal256)
+	if !ok {
+		return false
+	}
+	return s.Valid == r.Valid && s.Value == r.Value
+}
+
+func (s *Decimal256) String() string {
+	if !s.Valid {
+		return nullValueStr
+	}
+	return s.DataType().String() + " value"
```

Comment:
do we need ` + " value"` & why?

Reply:
https://github.com/cloudquery/plugin-sdk/pull/914#discussion_r1210098302